### ISSUE TITLE
Implement `hardhat_setChainId`

### ISSUE THREAD

**Issue by alcuadrado** [NomicFoundation/hardhat#3074]
We should implement this method, as it's useful for testing things like EIP-712 related functionality

**Comment by alcuadrado** [comment]
This comes from a conversation with Ryan about https://github.com/NomicFoundation/hardhat/pull/3060 — I think `hardhat_setChainId` would be a better alternative to #3060

**Comment by nataouze** [comment]
Agreed with having a `hardhat_setChainId`. EIP-712 is pretty common and it would be useful to have tests for network-fork-proofing.

**Comment by schekhovskoi-hacken** [comment]
Might be useful for [EIP-6551](https://eips.ethereum.org/EIPS/eip-6551) testing
